astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
+        data = hdus[ext_data].data
+        if dtype is not None:
+            data = data.astype(dtype)
+        meta = hdus[ext_meta].header
+
+        # Read the mask and uncertainty from the specified extensions but
+        # silently fail if the extension does not exist.
+        mask = None
+        if ext_mask in hdus:
+            mask = hdus[ext_mask].data
+            # Convert it to boolean array?
+            if kw_hdr_masktype in hdus[ext_mask].header.get('comment', []):
+                mask = mask.astype(bool)
+
+        uncertainty = None
+        if ext_uncert in hdus:
+            uncertainty = hdus[ext_uncert].data
+
+            hdr = hdus[ext_uncert].header
+            # Get the required class for the uncertainty
+            cls = (kw_hdr_uncerttype[kw] for kw in kw_hdr_uncerttype
+                   if kw in hdr.get('comment', []))
+            cls = next(cls, UnknownUncertainty)
+
+            # Get the unit for the uncertainty if present
+            unit_ = hdr[kw_unit].lower() if kw_unit in hdr else None
+
+            # Don't copy it here, if a copy is required do it when creating
+            # NDData.
+            uncertainty = cls(uncertainty, unit=unit_, copy=False)
+
+        # Load unit and wcs from header
+        unit = None
+        if kw_unit is not None and kw_unit in meta:
+            try:
+                unit = u.Unit(meta[kw_unit])
+            except ValueError:
+                # ValueError is raised if the unit isn't convertible to an
+                # astropy unit. Maybe they tried all-uppercase: maybe lowercase
+                # will work
+                unit = u.Unit(meta[kw_unit].lower())
+        wcs = WCS(meta)
+
+    # Just create an NDData instance: This will be upcast to the appropriate
+    # class
+    return NDData(data, meta=meta, mask=mask, uncertainty=uncertainty,
+                  wcs=wcs, unit=unit, copy=copy)
+
+
+def write_data_fits(ndd, filename, ext_mask='mask', ext_uncert='uncert',
+                    kw_unit='bunit', **kwargs_for_write):
+    """
+    Take an `~astropy.nddata.NDData`-like object and save it as FITS file.
+
+    Parameters
+    ----------
+    ndd : `astropy.nddata.NDData`-like
+        The data which is to be saved. Must not be given when this function
+        is called through the ``NDData.write``-method!
+
+    filename : str
+        The filename for the newly written file.
+
+    ext_mask, ext_uncert : str or int, optional
+        Extensions to which ``mask`` and ``uncertainty`` are written.
+        Default is ``'mask'`` (mask) and ``'uncert'`` (uncertainty).
+
+    kwargs_for_write :
+        Additional keyword arguments that are passed to
+        :func:`~astropy.io.fits.HDUList.writeto` (not all of them might be
+        possible).
+
+    Notes
+    -----
+    The ``data`` and ``meta`` are always written to the PrimaryHDU (extension
+    number ``0``).
+    """
+    # Comment card strings to allow roundtripping (must be identical to read!)
```

Comment:
I'd be inclined to save these in keywords rather than comments...though that brings up the the question of *which* keywords.